src/utils/utils.cpp

```diff
@@ -345,6 +348,197 @@ void PrintValue(std::ostream &os, const mg_value *value) {
 
 } // namespace utils
 
+namespace {
+// Completion and syntax highlighting support
+
+std::vector<std::string> GetCompletions(const char *text) {
+  std::vector<std::string> matches;
+
+  // Collect a vector of matches: vocabulary words that begin with text.
+  std::string text_str = utils::ToUpperCase(std::string(text));
+  for (auto word : constants::kCypherKeywords) {
+    if (word.size() >= text_str.size() &&
+        word.compare(0, text_str.size(), text_str) == 0) {
+      matches.emplace_back(word);
+    }
+  }
+  for (auto word : constants::kMemgraphKeywords) {
+    if (word.size() >= text_str.size() &&
+        word.compare(0, text_str.size(), text_str) == 0) {
+      matches.emplace_back(word);
+    }
+  }
+  for (auto word : constants::kAwesomeFunctions) {
+    if (word.size() >= text_str.size() &&
+        word.compare(0, text_str.size(), text_str) == 0) {
+      matches.emplace_back(word);
+    }
+  }
+
+  return matches;
+}
+
+void AddCompletions(replxx_completions *completions, const char *text) {
+  auto text_completions = GetCompletions(text);
+  for (const auto &completion : text_completions) {
+    replxx_add_completion(completions, strdup(completion.c_str()));
+  }
+}
+
+// taken from the replxx example.c (correct handling of UTF-8)
+int utf8str_codepoint_length(char const *s, int utf8len) {
+  int codepointLen = 0;
+  unsigned char m4 = 128 + 64 + 32 + 16;
+  unsigned char m3 = 128 + 64 + 32;
+  unsigned char m2 = 128 + 64;
+  for (int i = 0; i < utf8len; ++i, ++codepointLen) {
+    char c = s[i];
+    if ((c & m4) == m4) {
+      i += 3;
+    } else if ((c & m3) == m3) {
+      i += 2;
+    } else if ((c & m2) == m2) {
+      i += 1;
+    }
+  }
+  return codepointLen;
+};
+
+static char const wb[] = " \t\n\r\v\f-=+*&^%$#@!,./?<>;:`~'\"[]{}()\\|";
+
+int context_length(char const *prefix) {
+  // word boundary chars
+  int i = (int)strlen(prefix) - 1;
```

Comment:
C style casts are fine but in my opinion, they can lead to surprising behavior because of [this](https://en.cppreference.com/w/cpp/language/explicit_cast) (read explanation 1) so I would recommend a `static_cast` instead

Reply:
```suggestion
  int i = static_cast<int>(strlen(prefix)) - 1;
```